akka-stream/src/main/scala/akka/stream/impl/ActorRefBackpressureSinkStage.scala

```diff
@@ -52,10 +52,9 @@ private[akka] class ActorRefBackpressureSinkStage[In](ref: ActorRef, onInitMessa
       }
 
       private def sendData(): Unit = {
-        if (!buffer.isEmpty) {
-          ref ! buffer.poll()
-          acknowledgementReceived = false
-        }
+        if (buffer.size() == maxBuffer) tryPull(in)
```

Comment:
Yes, it frees up a slot, and hence it *unstucks* the upstream (this was a `pushAndPull` in the old interpreter). What got confused me here for a second though is that `.poll()` actually removes the element.